source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -18,109 +20,197 @@ minimize the objective function
 where :math:`d(x_i, C)` is the distance from :math:`x_i` to the closest center
 in :math:`C`.
 
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::kmeans::model run_training(const onedal::table& data,
+                                      const onedal::table& initial_centroids) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(10)
+         .set_max_iteration_count(50)
+         .set_accuracy_threshold(1e-4);
+
+      const auto result = onedal::train(kmeans_desc, data, initial_centroids);
+
+      print_table("labels", result.get_labels());
+      print_table("centroids", result.get_model().get_centroids());
+      print_value("objective", result.get_objective_function_value());
+
+      return result.get_model();
+   }
+
+   onedal::table run_inference(const onedal::kmeans::model& model,
+                               const onedal::table& new_data) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(model.get_cluster_count());
+
+      const auto result = onedal::infer(kmeans_desc, model, new_data);
+
+      print_table("labels", result.get_labels());
+   }
+
+
 ---
 API
 ---
 
 Descriptor
 ----------
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+::
 
    namespace method {
-   struct lloyd {};
-   using by_default = lloyd;
+      struct lloyd {};
+      using by_default = lloyd;
    } // namespace method
 
    template <typename Float = float,
```

Comment:
This will be part of coding guidelines, TBB come with the same question. `PascalCase` for template parameters. Parameter order is the same as in DAAL.